### ISSUE TITLE
Nesting complications

### ISSUE THREAD

**Issue by IvenBach** [rubberduck-vba/Rubberduck#3444]
Per @retailcoder: 'We need to be able to statically determine how to simplify that. Like [#3363](https://github.com/rubberduck-vba/Rubberduck/issues/3363) but a step or two further'

**Comment by bclothier** [comment]
Just to point out some test cases that may be helpful.

This generally shouldn't be a problem:

    Select Case Foo
      Case 1
        DoThis
     Case 2
       DoThat
     Case 3
       Select Case Bar
          Case "a"
            Durr
          Case "b"
            Doh
          Case Else
            Daww
        End Select
    End Select

OTOH, this definitely smells:

    Select Case Foo
      Case 1
        Select Case Bar
          Case "a"
            DoThat
          Case "b"
            DoThis
          Case Else
            Daww
       End Select
     Case 2
       Select Case Bar
          Case "a"
            DoThis
          Case "b"
            DoThat
          Case Else
            Daww
       End Select
     Case 3
       Select Case Bar
          Case "a"
            Durr
          Case "b"
            Doh
          Case Else
            Daww
        End Select
      Case Else
         GiveUp
    End Select

Note that the 2nd condition need not be repeated for all cases; having the condition duplicated in 2 branch should be sufficient to trigger the inspection as it violates the DRY rule.

But this might be actually OK, though it can be argued that there are other quality problems with the code:

    Select Case Foo
      Case 1
        Select Case Bar
          Case "a"
            DoThat
          Case "b"
            DoThis
          Case Else
            Daww
       End Select
     Case 2
       Select Case NotBar
          Case 0
            DoThis
          Case 1
            DoThat
          Case Else
            Daww
       End Select
     Case 3
       Select Case SomeThingElse
          Case "z"
            Doh
          Case "y"
            Durr
          Case Else
            Daww
        End Select
    End Select

The example doesn't technically repeat anything, though the logic is likely unnecessarily hard to follow but I think that's a different scope for a different inspection (maybe "TooMuchConditions" inspection?).
